build/RuntimeStore.targets

```diff
@@ -0,0 +1,176 @@
+  </Target>
+
+  <Target Name="UpdateNuGetConfig">
+    <UpdatePackageSource
+      NuGetConfigPath="$(RepositoryRoot)NuGet.config"
+      SourceName="Dependencies"
+      SourceUri="$(_DependencyBuildDirectory)"
+      Condition="Exists('$(_DependencyBuildDirectory)')" />
+  </Target>
+
+  <Target Name="_PrepareRuntimeStoreBuildAssets">
+    <!-- Clear working directory -->
+    <RemoveDir Directories="$(_WorkRoot)" />
+
+    <!-- Copy and update build assets -->
+    <Copy SourceFiles="$(RepositoryRoot)NuGet.config" DestinationFiles="$(_WorkRoot)NuGet.config" />
+    <Copy SourceFiles="$(_TemplatesDirectory)RS.Manifest\RS.Manifest.csproj" DestinationFiles="$(_WorkRoot)RS.Manifest.csproj" />
+  </Target>
+
+  <Target Name="_ResolveRuntimeStoreRID">
+    <Error Text="Please specify the platform for the runtime package store via RUNTIMESTORE_PLATFORM: {x64|x86}." Condition="'$(RUNTIMESTORE_PLATFORM)' == ''"/>
+
+    <GetOSPlatform>
+      <!-- Returns {Linux, macOS, Windows} -->
+      <Output TaskParameter="PlatformName" PropertyName="OSPlatform" />
+    </GetOSPlatform>
+
+    <PropertyGroup>
+      <RID Condition="'$(OSPlatform)' == 'Windows'">win7-$(RUNTIMESTORE_PLATFORM)</RID>
+      <RID Condition="'$(OSPlatform)' == 'Linux'">linux-$(RUNTIMESTORE_PLATFORM)</RID>
+      <RID Condition="'$(OSPlatform)' == 'macOS'">osx-$(RUNTIMESTORE_PLATFORM)</RID>
+    </PropertyGroup>
+  </Target>
+
+  <Target Name="BuildRuntimeStore" DependsOnTargets="UpdateNuGetConfig;_ResolveRuntimeStoreRID;_PrepareRuntimeStoreBuildAssets;_ResolveCurrentSharedFrameworkVersion">
+    <!-- Build runtime store -->
+    <Exec Command="dotnet store -m $(_WorkRoot)RS.Manifest.csproj -f netcoreapp2.0 -r $(RID) -o $(_RuntimeStoreOutputDirectory) --framework-version 2.0.2-* -w $(_RuntimeStoreWorkDirectory) $(RUNTIMESTORE_ADDITIONAL_ARGS)" />
+  </Target>
+
+  <Target Name="_ResolveCurrentSharedFrameworkVersion">
+    <!-- Parse framework version -->
+    <Exec Command="powershell.exe $(_BuildScriptsDirectory)GetSharedFrameworkVersion.ps1" ConsoleToMSBuild="true" Condition="'$(OS)' == 'Windows_NT'">
+      <Output TaskParameter="ConsoleOutput" PropertyName="SharedFrameworkVersion" />
+    </Exec>
+    <Exec Command="bash $(_BuildScriptsDirectory)GetSharedFrameworkVersion.sh" ConsoleToMSBuild="true" Condition="'$(OS)' != 'Windows_NT'">
+      <Output TaskParameter="ConsoleOutput" PropertyName="SharedFrameworkVersion" />
+    </Exec>
+  </Target>
+
+  <Target Name="BuildHostingDeps" DependsOnTargets="ResolveRepoInfo;_ResolveCurrentSharedFrameworkVersion">
+    <!-- Generate RS.Reference project -->
+    <RepoTasks.ResolveHostingStartupPackages
+      BuildArtifacts="@(ArtifactInfo)"
+      PackageArtifacts="@(PackageArtifact)">
+      <Output TaskParameter="HostingStartupArtifacts" ItemName="HostingStartupArtifacts" />
+    </RepoTasks.ResolveHostingStartupPackages>
+
+    <!--- MSBuild caches things if you run inproc so have to use Exec -->
```

Comment:
What are you having trouble cache-busting? inproc MSBuild calls are ideal because it preserves logging, parameters, etc.

Reply:
I'm not sure. I copied @pakrym's code here

Reply:
We use same project file to restore and build deps file per hosting startup, so we were cache busting restore targets and build targets when using same csproj but different dependency.

Reply:
Ok. You can cache-bust MSBuild's stored result by passing in new properties.

Reply:
I tried, we are passing lots of things as properties, it didn't work correctly.